gateway_test.go

```diff
@@ -270,57 +292,132 @@ var testsFallthrough = []FallthroughCase{
 	},
 }
 
-var testServiceIndexes = map[string][]netip.Addr{
-	"svc1.ns1":         {netip.MustParseAddr("192.0.1.1"), netip.MustParseAddr("fd12:3456:789a:1::")},
```

Comment:
Should be done using netip, and not reinventing the wheel. There's a reason for using that library.

Reply:
Look at the testDNSEndpointIndexes. You'll see the logic.